types/unified-tests.ts

```diff
@@ -27,19 +27,97 @@ const settings = {
   random: 'option'
 }
 
+interface ExamplePluginSettings {
+  example: string
+}
+const typedPlugin: Plugin<ExamplePluginSettings> = function() {}
+const typedSetting = {example: 'example'}
+
+const implicitlyTypedPlugin = (settings?: ExamplePluginSettings) => {}
+
+const pluginWithTwoSettings = (
+  processor?: Processor,
+  settings?: ExamplePluginSettings
+) => {}
+
 processor.use(plugin)
 processor.use(plugin).use(plugin)
+processor.use(plugin, settings)
+processor.use([plugin, plugin])
+processor.use([plugin])
+processor.use([plugin, settings])
+processor.use([[plugin, settings], [plugin, settings]])
+
+processor.use(typedPlugin)
+processor.use(typedPlugin).use(typedPlugin)
+processor.use(typedPlugin, typedSetting)
+processor.use([typedPlugin, typedSetting])
+processor.use([[typedPlugin, typedSetting], [typedPlugin, typedSetting]])
+processor.use([[plugin, settings], [typedPlugin, typedSetting]])
+processor.use([typedPlugin])
+
+processor.use(implicitlyTypedPlugin)
+processor.use(implicitlyTypedPlugin).use(implicitlyTypedPlugin)
+processor.use(implicitlyTypedPlugin, typedSetting)
+processor.use([implicitlyTypedPlugin, typedSetting])
+processor.use([
+  [implicitlyTypedPlugin, typedSetting],
+  [implicitlyTypedPlugin, typedSetting]
+])
+processor.use([[plugin, settings], [implicitlyTypedPlugin, typedSetting]])
+processor.use([implicitlyTypedPlugin])
+
+// NOTE: settings overrides the generic undefined
+// settings value will be unused but TypeScript will not warn
+processor.use(implicitlyTypedPlugin, typedSetting, settings)
```

Comment:
One very minor drawback of having two generic params, happens when an implicitly typed plugin with one param, is passed a second param, TypeScript doesn't know it should warn that the value is unused.
:notebook: It won't cause any breakage in code, it just leaves the unused value hanging.

Reply:
I think it is fine too.